File: Native/iOS/LunarConsoleApp/LunarConsoleApp/FakeLogEntry.m
//
//  FakeLogEntry.m
//  LunarConsoleApp
//
//  Created by Alex Lementuev on 11/23/15.
//  Copyright © 2015 Space Madness. All rights reserved.
//

#import "FakeLogEntry.h"

@implementation FakeLogEntry

- (instancetype)initWithType:(LUConsoleLogType)type message:(NSString *)message stackTrace:(NSString *)stackTrace
{
    self = [super init];
    if (self)
    {
        _type = type;
        _message = [message copy];
        _stacktrace = [stackTrace copy];
    }
    return self;
}

- (void)dealloc
{
    LU_RELEASE(_message);
    LU_RELEASE(_stacktrace);
    LU_SUPER_DEALLOC
}

@end

File: Native/iOS/LunarConsoleApp/LunarConsoleApp/FakeLogEntry.swift
//
//  FakeLogEntry.swift
//  LunarConsoleApp
//
//  Created by Alex Lementuev on 11/25/16.
//  Copyright © 2016 Space Madness. All rights reserved.
//

import CoreFoundation

class FakeLogEntry {
    
    private(set) var type: LUConsoleLogType
    private(set) var message: String
    private(set) var stacktrace: String?
    
    init(type: LUConsoleLogType, message: String, stacktrace: String?) {
        self.type = type
        self.message = message
        self.stacktrace = stacktrace
    }
}
